src/main/java/com/kodilla/ecommercee/domain/Cart.java

```diff
@@ -0,0 +1,61 @@
+package com.kodilla.ecommercee.domain;
+
+import com.kodilla.ecommercee.user.User;
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+import java.util.ArrayList;
+import java.util.List;
+
+
+@Entity
+@Table(name="carts")
+public class Cart {
+    private long id;
+    private User user;
+    private List<Product> products;
+
+    public Cart(long id, User user) {
+        this.id = id;
+        this.user = user;
+        this.products = new ArrayList<>();
+    }
+
+    public Cart() {
+    }
+
+    @GeneratedValue
+    @Id
+    @NotNull
+    @Column(name = "cart_id")
+    public long getId() {
+        return id;
+    }
+
+    @OneToOne(cascade = CascadeType.PERSIST, fetch = FetchType.EAGER)
+    @JoinColumn(name="user_id")
+    public User getUser() {
+        return user;
+    }
+
+    @ManyToMany(cascade = CascadeType.PERSIST)
+    @JoinTable(
+            name="join_product_cart",
+            joinColumns = {@JoinColumn(name="cart_id",referencedColumnName = "cart_id")},
+            inverseJoinColumns = {@JoinColumn(name = "product_id", referencedColumnName = "product_id")}
+    )
+    public List<Product> getProducts() {
+        return products;
+    }
```

Comment:
W Cart dodałeś ManyToMany to w Product  też trzeba to dodać (a później też w ProductDto). Pytanie tylko kto ma to zrobić czy ty czy ja? W każdym razie pytanie mam czy na pewno ManyToMany. Jak ktoś doda przedmiot do koszyka i go kupi, to następna osoba dodaje do koszyka może taki sam ale nie ten sam przedmiot. Może to też być np. książka ale nie ta sama sztuka... Ja bym chyba zrobiła OneToMany tutaj.

Reply:
nie zgodzę sie co do Many to MAny - do koszyka wrzuca się produkt jako produkt a nie sztukę, nie mamy przewidzianych kontroli stanów magazynowych czy czegoś w tym stylu. Moim zdaniem Many to Many - jeden koszyk ma wiele produktów, jeden produkt może być w wielu koszykach.

Reply:
Chociaż jak teraz sobie myślę, to bez sesnu mieć listę koszyków w klasie product.... Wystarczy ManyToOne, chyba masz jednak rację:)

Reply:
Jednak zostaje przy ManyToMany. Proszę o opinię.

Reply:
Zamiast bezargumentowego konstruktora dodaj @NoArgsConstructor

Reply:
Wyjdzie na to samo :)